src/request.rs

```diff
@@ -44,7 +47,24 @@ impl fmt::Display for Request {
             let values = values.join(",");
             writeln!(f, "{}: {}", name, values)?;
         }
-        writeln!(f, "{}", String::from_utf8_lossy(&self.body))
+
+        if self.body.len() > self.body_print_limit {
+            writeln!(
+                f,
+                "Body is too large to print: {} bytes (limit: {} bytes)",
+                self.body.len(),
+                self.body_print_limit
+            )?;
```

Comment:
I'd suggest doing something different here: we print _up to the limit_ and then say "We truncated the body because it was too large: {} bytes (limit: {} bytes)".